contracts/chain-adapters/ForwarderBase.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title ForwarderBase
+ * @notice This contract expects to receive messages and tokens from the Hub Pool on L1. It rejects messages which do not originate
+ * from the hub pool.
+ * @notice Base contract deployed some network between L1 and the network housing a spoke pool contract. If a message is sent
+ * to this contract which came from the cross domain admin, then it is routed to the next network via the canonical messaging
+ * bridge (which may be done automatically by the network's finalization infrastructure). Tokens sent from L1 are stored temporarily
+ * on this contract. Any EOA can initiate a bridge of these tokens to the target `l3SpokePool`.
+ * @custom:security-contact bugs@across.to
+ */
+abstract contract ForwarderBase is UUPSUpgradeable {
+    // L3 address of the recipient of L1 messages and tokens.
```

Comment:
```suggestion
    // L3 address of the recipient of L1 messages sent via the re-router adapter on L1 and tokens.
```

Reply:
designed to be deployed on L2 to re-route messages from L1 to L3